practice/rain.py

```diff
@@ -4,139 +4,99 @@
 # 1. Setup
 import operator
 
+
+# 2. Define
+
 def import_table():
-    with open('sunnysiderain.txt', 'r') as rain_table:
-        initial_list = rain_table.readlines()
-    return initial_list
+    """This imports the table to be read from the file"""
+    with open('sunnysiderain.txt') as rain_table:
+        intial_table = rain_table.readlines()
+    stripped_table = strip_table(intial_table)
+    return cut_out_items(stripped_table)
 
 def strip_table(table):
-    r"""
+    r"""Strips the table of formatting.
+
     >>> strip_table([''] * 12 + ['A \n'])
     [['A']]
-
-    :param table:
-    :return:
     """
-    stripped_table = [line.strip().split() for line in table[12:]]
-    return stripped_table
+    return [line.strip().split() for line in table[12:]]
 
 def cut_out_items(stripped_table):
-    """
+    """Cuts out unused parts of the imported data.
+
     >>> cut_out_items([['A', 'B', 'C']])
     [['A', 'B']]
-
-    :param stripped_table:
-    :return:
     """
-    cut_table = [line[:2] for line in stripped_table]
-    return cut_table
+    return [line[:2] for line in stripped_table]
 
-def make_lookup_table(formatted_table):
-    """
-    >>> make_lookup_table([['A', 'B']])
-    {'A': 'B'}
+def get_date_to_amt(table):
+    """Turns data into a dictionary that pairs each date with the rainfall amount on that day.
 
-    :param formatted_table:
-    :return:
+    >>> get_date_to_amt([['A', '3']])
+    {'A': 3}
     """
-    lookup_table = {key: value for (key, value) in formatted_table}
-    return lookup_table
+    date_to_amt = {key: value for (key, value) in table}
```

Comment:
If your data is already formatted like `[(key, val), (key, val)...]` then you can just use `dict()`. Little trick.